toolbox/build_info.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+#
+# Helper to get build information, including OS, platform, CPU arch, compiler info, and Docker version...
+#
+#########################################################################################
+#
+# CelestialCalendar Automation:
+#   Python automation scripts for building and testing the CelestialCalendar C++ project.
+# 
+# Author : Ningqi Wang (0xf3cd)
+# Email  : nq.maigre@gmail.com
+# Repo   : https://github.com/0xf3cd/celestial-calendar
+# License: GNU General Public License v3.0
+# 
+# This software is distributed without any warranty.
+# See <https://www.gnu.org/licenses/> for more details.
+
+import os
+import sys
+import json
+import platform
+import argparse
+
+from pathlib import Path
+from datetime import datetime, UTC
+from dataclasses import dataclass, asdict
+
+from typing import Optional, List, Sequence
+
+
+# Apply a workaround to import from the parent directory...
+sys.path.append(str(Path(__file__).parent.parent))
+
+from automation import (
+  run_cmd, ProcReturn, yellow_print, green_print
+)
+
+
+def silent_run(cmd: Sequence[str], **kwargs) -> ProcReturn:
+  '''Run a command without printing anything.'''
+  return run_cmd(cmd, print_cmd=False, print_stdout=False, print_stderr=False, **kwargs)
+
+
+@dataclass(frozen=True)
```

Comment:
Maybe add a new field `additional_info`. On linux, we can use `lscpu` or `cat /proc/cpuinfo` to retrieve additional CPU info; on macOS and windows, we have other commands to get the additional CPU info.

Reply:
Not sure if this is really helpful. But getting more info of the building environment is not harmful, right?

Reply:
Also, use `libelf`/`readelf` on linux, `dumpbin` on windows to inspect the shared libs, `otool` on macOS